protocol/contracts/oud/OudRedeemer.sol

```diff
@@ -0,0 +1,145 @@
+  uint256 public constant tpiDecimals = 4;
+
+  // Target address for 'stable' token used for redemption of Oud + stable for Temple
+  address public depositStableTo;
+
+  // @dev requires this contract to have minting rights for both Temple and Oud
+  constructor(
+    address _oudToken,
+    address _templeToken,
+    address _stable,
+    address _depositStableTo,
+    uint256 _tpi
+  ) {
+    templeToken = _templeToken;
+    oudToken = _oudToken;
+    stable = IERC20(_stable);
+    depositStableTo = _depositStableTo;
+    tpi = _tpi;
+  }
+
+  //  @notice Returns the index at which 'stable' token is required to mint Temple
+  function treasuryPriceIndex() external view returns (uint256) {
+    return tpi;
+  }
+
+  // @dev Allows the tpi to be adjusted.
+  // @param 'value' is the new TPI
+  function setTreasuryPriceIndex(uint256 value) external override onlyOwner {
+    tpi = value;
+  }
+
+  function setStableCoin(address newStable) external override onlyOwner {
+    if (newStable == address(0)) revert AddressZero();
+    stable = IERC20(newStable);
+  }
+
+  // @dev Decimal precision for 'tpi', 9880 == $0.988, precision = 4
+  function TPI_PRECISION() external pure returns (uint256) {
+    return tpiDecimals;
+  }
+
+  /**
+   * @notice Provides a quote for 'oudAmount' with relevant token addresses:
+   *  'oudAmount' provided, 'stable' token required, 'templeAmount' that will be minted to caller
+   * @param oudAmount amount of Oud that will be redeemed
+   * @return _oudAmount amount of Oud that wil be redeemed
+   * @return _oudTokenAddress address of the Oud token
+   * @return _stableAmount amount of Stable token required at tpi
+   * @return _stableAddress address of the Stable token in use
+   * @return _templeAmount amount of Temple that will be minted
+   * @return _templeTokenAddress address of the Temple token
+   */
+  function getQuote(
+    uint256 oudAmount
+  )
+    external
+    view
+    returns (
+      uint256 _oudAmount,
+      address _oudTokenAddress,
+      uint256 _stableAmount,
+      address _stableAddress,
+      uint256 _templeAmount,
+      address _templeTokenAddress
+    )
+  {
+    return (
+      oudAmount,
+      oudToken,
+      _getStableAmount(oudAmount),
+      address(stable),
+      oudAmount,
+      templeToken
+    );
+  }
+
+  // @notice Redeems Oud and Stable token (@ tpi) for Temple
+  // @dev requires allowances for Oud and Stable Token
+  // @param 'oudAmount' The amount of Oud to be redeemed for Temple
+  // @returns 'templeAmount' The amount of Temple that has been minted to sender.
+  function redeem(uint256 oudAmount) external returns (uint256 templeAmount) {
+    if (oudAmount == 0) revert RedeemAmountZero();
+    address account = msg.sender;
+    uint256 _stableAmount;
+    _stableAmount = _getStableAmount(oudAmount);
+    IOudToken(oudToken).burn(account, oudAmount);
+    SafeERC20.safeTransferFrom(stable, account, address(this), _stableAmount);
```

Comment:
You have at the top of the contract
```
using SafeERC20 for IERC20;
```

The whole reason for that (which is considered best practice for libs) is that it allows you to condense these two lines into:
```
stable.safeTransferFrom(account, address(this), _stableAmount);
stable.safeIncreaseAllowance(depositStableTo, _stableAmount);
```

Reply:
[0895b66](https://github.com/TempleDAO/temple/pull/760/commits/0895b66c3800129fe1e10ec09bfc167fbd4513dc)